javatests/arcs/android/storage/service/StorageServiceNgImplTest.kt

```diff
@@ -0,0 +1,176 @@
+    val storageKey = RamDiskStorageKey("store1")
+    val encodedStoreOptions = createEncodedStoreOptions(
+      storageKey,
+      CountType(),
+      ParcelableCrdtType.Count
+    )
+
+    storageService.openStorageChannel(encodedStoreOptions, channelCallback, messageCallback)
+    val channel = channelCallback.waitForOnCreate() as StorageChannelImpl
+
+    assertThat(storageService.activeStorageKeys()).containsExactly(storageKey)
+    assertThat(channel.storageKey).isEqualTo(storageKey)
+  }
+
+  @Test
+  fun openingTwoChannels_forDifferentStores_createsTwoStores() = scope.runBlockingTest {
+    // Define the store options for two different stores.
+    val storageKey1 = RamDiskStorageKey("store1")
+    val encodedStoreOptions1 = createEncodedStoreOptions(
+      storageKey1,
+      CountType(),
+      ParcelableCrdtType.Count
+    )
+
+    val storageKey2 = RamDiskStorageKey("store2")
+    val encodedStoreOptions2 = createEncodedStoreOptions(
+      storageKey2,
+      CountType(),
+      ParcelableCrdtType.Count
+    )
+
+    // Open two storage channels for two distinct stores
+    storageService.openStorageChannel(encodedStoreOptions1, channelCallback, messageCallback)
+    val channel1 = channelCallback.waitForOnCreate() as StorageChannelImpl
+
+    val channelCallback2 = FakeStorageChannelCallback()
+    storageService.openStorageChannel(encodedStoreOptions2, channelCallback2, messageCallback)
+    val channel2 = channelCallback2.waitForOnCreate() as StorageChannelImpl
+
+    // Check two stores have been created.
+    assertThat(storageService.activeStorageKeys())
+      .containsExactlyElementsIn(listOf(storageKey1, storageKey2))
+
+    // Check the channels are communicating with the expected stores.
+    assertThat(channel1.storageKey).isEqualTo(storageKey1)
+    assertThat(channel2.storageKey).isEqualTo(storageKey2)
+  }
+
+  @Test
+  fun openingTwoChannels_forSameStore_onlyCreatesOneStore() = scope.runBlockingTest {
+    val storageKey = RamDiskStorageKey("store1")
+    val encodedStoreOptions = createEncodedStoreOptions(
+      storageKey,
+      CountType(),
+      ParcelableCrdtType.Count
+    )
+
+    storageService.openStorageChannel(encodedStoreOptions, channelCallback, messageCallback)
+    val channel1 = channelCallback.waitForOnCreate() as StorageChannelImpl
+
+    val channelCallback2 = FakeStorageChannelCallback()
+    storageService.openStorageChannel(encodedStoreOptions, channelCallback2, messageCallback)
+    val channel2 = channelCallback2.waitForOnCreate() as StorageChannelImpl
+
+    assertThat(storageService.activeStorageKeys()).containsExactly(storageKey)
+
+    assertThat(channel1.storageKey).isEqualTo(storageKey)
+    assertThat(channel2.storageKey).isEqualTo(storageKey)
+  }
+
+  private fun createEncodedStoreOptions(
+    storageKey: StorageKey,
+    type: Type,
+    crdtType: ParcelableCrdtType
+  ): ByteArray = StoreOptions(storageKey, type).toParcelToByteArray(crdtType)
```

Comment:
Since you always pass in `CountType` and `ParcelableCrdtType.Count`, you might as well just hard code them here. They aren't relevant for the tests, so it'll make the test methods simpler.

(also, too long to be an expression function)